components/MdInput.test.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import {
+  render,
+  screen,
+  fireEvent,
+  queryByTestId
+} from '@testing-library/react'
+import { MdInput } from './MdInput'
+
+describe('write inside textarea', () => {
+  test('renders text when user types', () => {
+    const { container } = render(<MdInput />)
+    fireEvent.change(screen.getByRole('textbox'), {
+      target: { value: 'Javascript' }
+    })
+    expect(screen.getByRole('textbox').value).toContain('Javascript')
+    expect(container).toMatchSnapshot()
+  })
+})
+
+describe('click Write Button', () => {
+  test('switches to write mode when user clicks Write button', () => {
+    const { container } = render(<MdInput />)
+    fireEvent.click(screen.getByText('Write'))
+    const textbox = queryByTestId(container, 'textbox')
```

Comment:
I believe you can swap out the global method `queryByTestId` with a method that is returned from `render(<MdInput>)`